js/app/interfaces/filter.js

```diff
@@ -60,6 +68,8 @@ const Filter = new Lang.Interface({
      */
     include: function (model) {
         let retval = this.include_impl(model);
+        if (retval && this._sub_filter)
```

Comment:
I don't think this is correct, because the first filter's `invert` now applies to the sub-filter. E.g. if you have `Filter.Featured(invert: true)` with `Filter.Unread(invert: false)` as the subfilter, expressing "not featured and unread", then this will also include articles that are featured and read, or not featured and read.